components/twitch-panel/twitch-panel.tsx

```diff
@@ -0,0 +1,69 @@
+import { useEffect, useState } from "react";
+import { Container, Group, Text, Title, useMantineColorScheme } from "@mantine/core";
+import { TwitchStream } from "../../src/coh3/coh3-types";
+import ChannelList from "./channel-list";
+
+declare global {
+  interface Window {
+    Twitch: any;
+  }
+}
+
+type Props = {
+  twitchStreams: TwitchStream[] | null;
+  error: Error | null;
+};
+const TwitchPanel = ({ twitchStreams, error }: Props) => {
+  const { colorScheme } = useMantineColorScheme();
+  const [player, setPlayer] = useState<any>();
+
+  useEffect(() => {
+    // this gate only be needed because of react strict mode running things twice
+    if (document.getElementById("twitch-script") !== null || twitchStreams === null) return;
+
+    const script = document.createElement("script");
+    script.src = "https://player.twitch.tv/js/embed/v1.js";
+    script.id = "twitch-script";
+    document.body.appendChild(script);
+
+    script.addEventListener("load", () => {
+      const embed = new window.Twitch.Embed("twitch-embed", {
+        width: 854,
+        height: 480,
+        channel: twitchStreams[0].user_login,
+        layout: "video",
+        autoplay: false,
+        theme: colorScheme,
+        // Only needed if this page is going to be embedded on other websites
+        parent: ["embed.example.com", "othersite.example.com"],
+      });
+
+      embed.addEventListener(window.Twitch.Embed.VIDEO_READY, () => {
+        const player = embed.getPlayer();
+        player.play();
+        setPlayer(player);
+      });
+    });
+  }, [twitchStreams, colorScheme]);
+
+  function handleChangeChannel(channel: string) {
+    player.setChannel(channel);
+    player.play();
+  }
+
+  return (
+    <Container size="xl">
```

Comment:
You can use conatiner size fluid https://mantine.dev/core/container/, we have made the home page bigger. It's gonna give you some space.  So maybe we can fit 4 streams on the right side? 

Also we can put this all into paper component which will get the feeling of card https://mantine.dev/core/paper/

Or maybe the paper should be on the index.tsx and here we can keep the container. In case we would like to re-use is somewhere else. 

We are also using tabler icons https://tablericons.com/ so you can get the twitch icon from there :)

Reply:
Thanks for taking a look :)

- I'll try out the container size fluid
- 4 cards could be good.
- I saw that paper component. The way you're saying with paper in index I think might be best. 
- On that note I can probably break up the player, and the channel selection, so that if we want just the player it is pretty easy to do. And then I can pass the channel as a prop to the player.
- Not sure how big we want the stream itself to be either. I just set it to the height and width that the example in the twitch docs used. And not sure how we would want to resize it for mobile. I don't think we'll be able to do that with just css.

I can probably get it to a nice point for review tomorrow sometime. Then we can take a look and see what changes we want.

Reply:
<img width="504" alt="image" src="https://user-images.githubusercontent.com/8086995/225263483-189b32da-4477-43b5-b8bd-7d1fb49020ba.png">
This is how it looks on aoeworld on mobile.

it's great that we can take inspiration from that site where it works nicely and we don't need to come up with our own :D

Reply:
Awesome, I pushed some small changes, but not ready for review yet fyi.